lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/TableExceptionTranslatorImpl.java

```diff
@@ -49,6 +57,18 @@ public RuntimeException translateException(Throwable exception) {
 			return new RuntimeException(exception);
 		}
 	}
+	
+	
+	String replaceConstraintNameWithConstraintClause(String message) {
+		Optional<String> constraintName = getConstraintViolationName(message);
+		if(constraintName.isPresent()) {
+			Optional<String> clause = connectionFactory.getFirstConnection().getConstraintClause(constraintName.get());

```

Comment:
Maybe getConstraintClause could cache the values for the constraint name since by this time the temporary table already visited it (and could populate the cache)